### PR TITLE
Translation issues

### PR DESCRIPTION
**Pull request by jdalsem**
Performance decreased when i introduced this language fix for simplecached langauges/js here:  https://github.com/Elgg/Elgg/commit/ab17ee54dd3e71f8bb230ec3c9a4368c492b7cd1 
reason is that i did a `->getInstalledTranslations()` which loads all languages. That takes some time. I changed it by adding a function to return all language codes and use those to register the simplecache views. That also fixed an issue where you could request a `js/languages/<lang_id>` that produced a error instead of a nice fallback to english.

### CODE REVIEW

File: views/default/js/languages.php [hunk 1]
```diff
@@ -22,6 +22,12 @@
 $all_translations = elgg_get_config('translations');
 $translations = $all_translations['en'];
 
+if ($language != 'en' && !isset($all_translations[$language])) {
+	// try to reload missing translations
```
Comment (mrclay) [inline views/default/js/languages.php]: Why wouldn't they have been loaded by this point?

Comment (jdalsem) [inline views/default/js/languages.php]: normal elgg only loads a users language and `en`, not all installed translations

File: engine/classes/Elgg/I18n/Translator.php [hunk 2]
```diff
@@ -382,5 +384,158 @@ function getMissingLanguageKeys($language) {
...
 		return false;
 	}
+	
+	/**
+	 * Returns an array of language codes.
+	 *
+	 * @return array
+	 */
+	function getAllLanguageCodes() {
+		return array(
```
Comment (mrclay) [inline engine/classes/Elgg/I18n/Translator.php]: How was this generated and how can make it super easy to be kept up-to-date?

Comment (jdalsem) [inline engine/classes/Elgg/I18n/Translator.php]: copied from the language file

Comment (mrclay) [inline engine/classes/Elgg/I18n/Translator.php]: Shouldn't we just use that file here?

Comment (jdalsem) [inline engine/classes/Elgg/I18n/Translator.php]: uh do not know what you are suggesting

Comment (mrclay) [inline engine/classes/Elgg/I18n/Translator.php]: You know what, when a new language is created we can update this. lol

**Comment by mrclay** [comment]
LGTM

**Pull request merged**